webapp/src/main/webapp/script.js

```diff
@@ -17,6 +17,39 @@
 window.onload=function(){
   document.forms['myForm'].addEventListener('submit', (event) => {
     event.preventDefault();
+
+    // Create loading bar while waiting for the validator's results 
+    const errorOutput = document.getElementById('errorOutput');
+    const loadingSpinner = document.createElement('div');
+    loadingSpinner.className = "spinner-border";
+    loadingSpinner.setAttribute('role','status');
+    const loadingText = document.createElement("span");
+    loadingText.className = "sr-only";
+    loadingText.innerText = "Loading...";
+    loadingSpinner.appendChild(loadingText);
+    
+    // Create heading 
+    const resultsHeading = document.createElement('h2');
+    resultsHeading.id = "resultHeading";
+    resultsHeading.innerText = 'Results';
+    
+    // Clear heading
+    if (!(typeof errorOutput.nextSibling.id === 'undefined') && errorOutput.nextSibling.id.localeCompare("resultHeading") == 0) {
+      errorOutput.parentNode.removeChild(errorOutput.nextSibling);
+    }
+    // Clear notice containers
+    errorOutput.innerHTML = '';
+    const error = document.getElementById('error');
+    error.innerHTML = '';
+    const warning = document.getElementById('warning');
+    warning.innerHTML = '';
+    const unimplementedNotices = document.getElementById('unimplementedNotices');
+    unimplementedNotices.innerHTML = '';
```

Comment:
just wondering, why are you clearing the notice containers here?

Reply:
If you upload for a second time it will just keeping adding notices to the page because you are doing "appendChild" and there will be duplicates. Found this out through testing the webpage.

Reply:
Ah nice